rtl/eclrtl/rtlfield.cpp

```diff
@@ -376,6 +508,11 @@ __int64 RtlPackedIntTypeInfo::getInt(const void * ptr) const
         return rtlGetPackedSigned(ptr);
 }
 
+double RtlPackedIntTypeInfo::getReal(const void * ptr) const
+{
+    return (double) getInt(ptr);
```

Comment:
because (real)(unsigned __int64)0x8000000000000) != (real)(__int64)0x8000000000000)
same for RtlTypeInfo::getReal()